File: 796. Rotate String.c
bool rotateString(char* s, char* goal) {

    size_t len = strlen(s);
    for (int i = 0; i < len; i++){
        if (strncmp(s+i, goal, len-i) == 0 && strncmp(s, goal+len-i, i) == 0){
            return true;
        }
    }

    return false;
    
}

File: 796. Rotate String.rs
impl Solution {
    pub fn rotate_string(s: String, goal: String) -> bool {
        let s_len = s.len();

        for i in 0..s_len {
            let tmp = format!("{}{}", &s[i..], &s[..i]);
            if tmp == goal {
                return true;
            }
        }
        return false;
    }
}